src/main/java/kubatech/api/implementations/KubaTechGTMultiBlockBase.java

```diff
@@ -0,0 +1,160 @@
+package kubatech.api.implementations;
+
+import static kubatech.api.Variables.ln2;
+import static kubatech.api.Variables.ln4;
+
+import com.gtnewhorizons.modularui.api.drawable.IDrawable;
+import com.gtnewhorizons.modularui.api.widget.Widget;
+import gregtech.api.gui.modularui.GT_UITextures;
+import gregtech.api.metatileentity.implementations.GT_MetaTileEntity_ExtendedPowerMultiBlockBase;
+import gregtech.api.metatileentity.implementations.GT_MetaTileEntity_Hatch;
+import gregtech.api.util.GT_ExoticEnergyInputHelper;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.function.Function;
+import net.minecraft.item.ItemStack;
+import net.minecraftforge.fluids.FluidStack;
+
+public abstract class KubaTechGTMultiBlockBase<T extends GT_MetaTileEntity_ExtendedPowerMultiBlockBase<T>>
+        extends GT_MetaTileEntity_ExtendedPowerMultiBlockBase<T> {
+
+    protected KubaTechGTMultiBlockBase(int aID, String aName, String aNameRegional) {
+        super(aID, aName, aNameRegional);
+    }
+
+    protected KubaTechGTMultiBlockBase(String aName) {
+        super(aName);
+    }
+
+    /**
+     * Enables infinite overclocking (will give more outputs with more energy past 1 tick)
+     * Currently doesn't support recipe inputs
+     * @return If this supports infinite overclock
+     */
+    protected boolean isOverclockingInfinite() {
+        return false;
+    }
+
+    /**
+     * @return The minimum amount of ticks this multiblock can overclock to
+     */
+    protected int getOverclockTimeLimit() {
+        return 1;
+    }
+
+    @Override
+    protected void calculateOverclockedNessMultiInternal(
+            long aEUt, int aDuration, int mAmperage, long maxInputVoltage, boolean perfectOC) {
+        calculateOverclock(aEUt, aDuration, maxInputVoltage, perfectOC);
+    }
+
+    /**
+     * @param aEUt Recipe EU/t
+     * @param aDuration Recipe duration (in ticks)
+     * @param maxInputVoltage The amount of energy we want to overclock to
+     * @param isPerfect Is this overclock perfect ?
+     * @return The amount of overclocks
+     */
+    protected int calculateOverclock(long aEUt, int aDuration, final long maxInputVoltage, boolean isPerfect) {
+        int tiers = (int) (Math.log((double) maxInputVoltage / (double) aEUt) / ln4);
```

Comment:
Maybe it would be worthwhile to use the new overclock calculator here. Though I don't think that has infinite OC rn. Though if you do want to use that I'm sure @BlueWeabo could add it.

Reply:
Maybe later